### PR TITLE
Fix for CLI-271

### PR DESCRIPTION
**Pull request by laeubi**
Add the posibility to use Options as a parameter for fetching values from the CommandLine.

### CODE REVIEW

File: src/main/java/org/apache/commons/cli/CommandLine.java [hunk 1]
```diff
@@ -246,6 +331,44 @@ public String getOptionValue(char opt, String defaultValue)
...
+     * the 2nd argument is the value. If the option has only one argument
+     * (<tt>-Dfoo</tt>) it is considered as a boolean flag and the value is
+     * <tt>"true"</tt>.
+     *
+     * @param option name of the option
+     * @return The Properties mapped by the option, never <tt>null</tt>
+     *         even if the option doesn't exists
+     * @since 1.4
+     */
+    public Properties getOptionProperties(Option option)
```
Comment (laeubi) [inline src/main/java/org/apache/commons/cli/CommandLine.java]: I just copied the String variant of this including the javadoc, so I assume it was changed some time in the past and the java doc was not changed.

File: src/main/java/org/apache/commons/cli/CommandLine.java [hunk 2]
```diff
@@ -246,6 +331,44 @@ public String getOptionValue(char opt, String defaultValue)
...
+                }
+                else if (values.size() == 1)
+                {
+                    // no explicit value, handle it as a boolean
+                    props.put(values.get(0), "true");
+                }
+            }
+        }
+
+        return props;
```
Comment (laeubi) [inline src/main/java/org/apache/commons/cli/CommandLine.java]: Why does it not belong to CLI271? Its an adoption of the previous String variant (what is now delegating to this method) that I adjusted a bit to remove the requirements of string matching.

File: src/test/java/org/apache/commons/cli/OptionTest.java [hunk 3]
```diff
@@ -71,6 +72,13 @@ public void testClone()
         assertEquals(0, a.getValuesList().size());
         assertEquals(2, b.getValues().length);
     }
+    
+    @Test
+    public void testHashCode() {
```
Comment (laeubi) [inline src/test/java/org/apache/commons/cli/OptionTest.java]: Can you explain how I remove this from the PR? Must I create a new commit/PR?

File: src/main/java/org/apache/commons/cli/CommandLine.java (hunk 1 above)
Comment (britter) [inline src/main/java/org/apache/commons/cli/CommandLine.java]: Okay, I see. Makes sense make this symmetric with the String param based version.

File: src/main/java/org/apache/commons/cli/CommandLine.java (hunk 2 above)
Comment (britter) [inline src/main/java/org/apache/commons/cli/CommandLine.java]: I see, thank you for the clarification.

File: src/test/java/org/apache/commons/cli/OptionTest.java (hunk 3 above)
Comment (britter) [inline src/test/java/org/apache/commons/cli/OptionTest.java]: Since, I have to apply this via SVN, I can remove it for you.

**Pull request closed without merging**